test/integration/render_task_test.rb

```diff
@@ -0,0 +1,266 @@
+        ---
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod2
+        spec:
+          restartPolicy: Never
+          activeDeadlineSeconds: 60
+          containers:
+          - name: pod2
+            image: busybox
+            args:
+            - echo
+            - log from pod2
+        ---
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: config-for-pod2
+        data:
+          supports_partials: yep
+      RENDERED
+    end
+  end
+
+  def test_render_task_rendering_all_files
+    render = build_render_task(fixture_path('hello-cloud'))
+
+    assert_render_success render.run(mock_output_stream, [])
+    stdout_assertion do |output|
+      assert_match(/name: bare-replica-set/, output)
+      assert_match(/name: hello-cloud-configmap-data/, output)
+      assert_match(/name: ds-app/, output)
+      assert_match(/kind: PodDisruptionBudget/, output)
+      assert_match(/name: hello-job/, output)
+      assert_match(/name: redis/, output)
+      assert_match(/name: role-binding/, output)
+      assert_match(/name: resource-quotas/, output)
+      assert_match(/name: build-robot/, output)
+      assert_match(/name: stateful-busybox/, output)
+      assert_match(/name: hello-cloud-template-runner/, output)
+      assert_match(/name: unmanaged-pod-\w+/, output)
+      assert_match(/name: web/, output)
+    end
+  end
+
+  def test_render_invalid_binding
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a', 'b': 'binding-b')
+    fixture = 'deployment.yaml.erb'
+
+    assert_render_failure render.run(mock_output_stream, [fixture])
+    assert_logs_match_all([
+      /Invalid template: .*deployment.yaml.erb/,
+      /undefined local variable or method `supports_partials'/
+    ])
+  end
+
+  def test_render_invalid_arguments
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a')
+
+    assert_render_failure render.run(mock_output_stream, ["../"])
+    assert_logs_match_all([
+      %r{test/fixtures" is not a file}
+    ])
+  end
+
+  def test_render_path_outside_template_dir
+    render = build_render_task(fixture_path('test-partials'), 'a': 'binding-a')
+
+    assert_render_failure render.run(mock_output_stream, ["../hello-cloud/configmap-data.yml"])
+    assert_logs_match_all([
+      %r{test/fixtures/hello-cloud/configmap-data.yml" is outside the template dir}
+    ])
+  end
+
+  def test_render_empty_template_dir
+    render = build_render_task(Dir.mktmpdir)
+
+    assert_render_failure render.run(mock_output_stream)
+    assert_logs_match_all([
+      /no templates found in template dir/
+    ])
+  end
+
+  def test_render_invalid_yaml
+    render = build_render_task(fixture_path('invalid'))
+    fixture = 'yaml-error.yml'
+
+    assert_render_failure render.run(mock_output_stream, [fixture])
+    assert_logs_match_all([
+      /Invalid template: .*yaml-error.yml/,
+      /mapping values are not allowed/
```

Comment:
Is ` (<unknown>):` intentional in the error message:
`Error message: (<unknown>): mapping values are not allowed in this context at line 9 column 21`

Reply:
It is not, but that's from Psych. I added a commit to address this, it changes the output from looking like `(<unknown>)` to `(<rendered> the-file-in-question.yaml.erb)`